plugins/kubernetes/app/models/kuber_deploy_service.rb

```diff
@@ -60,4 +60,11 @@ def log(msg, extra_info = {})
 
     Kubernetes::Util.log msg, extra_info
   end
+
+  # Starts a new deploy watcher, forcing it to get in synch with the cluster
+  def start_watching_cluster
+    watcher = Celluloid::Actor[project.deploy_watcher_actor_name]
+    watcher.terminate if watcher and watcher.alive?
+    Celluloid::Actor[project.deploy_watcher_actor_name] = Watchers::DeployWatcher.new(project)
```

Comment:
I actually gave it another thought and started supervising it. We can restart it, anyway, and having it supervised protects us from any unexpected crash, which is good if we're in the middle of a deployment: if it crashes, Celluloid will bring it back, the watcher will get in-sync with the cluster and proceed with the deployment.

So, good point in here. Thanks.